docs/asciidoc/static/managing-multiline-events.asciidoc

```diff
@@ -0,0 +1,116 @@
+[[multiline]]
+=== Managing Multiline Events
+
+Several use cases generate events that span multiple lines of text. In order to correctly handle these multline events, 
+Logstash needs to know how to tell which lines are part of a single event.
+
+Whenever possible, avoid using multiline processing at all. When multiline processing is required, implement the
+processing as early in the pipeline as possible. The preferred tool in the Logstash pipeline is the 
+{logstash}plugins-codecs-multiline.html[multiline codec], which merges lines from a single input using a simple set of 
+rules.
+
+For more complex needs, the {logstash}plugins-filters-multiline.html[multiline filter] performs a similar task at the 
+filter stage of processing, where the Logstash instance aggregates multiple inputs.
+
+The most important aspects of configuring either multiline plugin are the following:
+
+* The `pattern` option specifies a regular expression. Lines that match the specified regular expression are considered 
+either continuations of a previous line or the start of a new multiline event. You can use 
+{logstash}plugins-filters-grok.html[grok] regular expression templates with this configuration option.
+* The `what` option takes two values: `previous` or `next`. The `previous` value specifies that lines that match the 
+value in the `pattern` option are part of the previous line. The `next` value specifies that lines that match the value 
+in the `pattern` option are part of the following line.* The `negate` option applies the multiline codec to lines that 
+_do not_ match the regular expression specified in the `pattern` option.
+
+See the full documentation for the {logstash}plugins-codecs-multiline.html[multiline codec] or the 
+{logstash}plugins-filters-multiline.html[multiline filter]plugin for more information on configuration options.
+
+==== Multiline Special Cases
+
+* The multiline codec plugin does not suport the {logstash}plugins-inputs-lumberjack[lumberjack] input plugin.
+
+* The multiline codec plugin does not support file input from files that contain events from multiple sources.
+
+* The multiline filter plugin is not thread-safe. Avoid using multiple filter workers with the multiline filter.
```

Comment:
how about 'using multiple filter workers introduces the possibility of losing line ordering, resulting in processing errors that can be very difficult to diagnose.'?